components/calendar/src/calendar.js

```diff
@@ -0,0 +1,368 @@
+import { colors, spacers } from '@dhis2/ui-constants'
+import { IconChevronLeft16, IconChevronRight16 } from '@dhis2/ui-icons'
+import cx from 'classnames'
+import { useDatePicker } from 'multi-calendar-dates'
+import PropTypes from 'prop-types'
+import React, { useState } from 'react'
+
+export const Calendar = ({
+    onDateSelect,
+    calendar,
+    initialDate,
+    dir = 'ltr',
+    locale = 'en',
+    numberingSystem = 'geor',
+    showSelectedDate = false,
+    weekDayFormat = 'narrow',
+    timeZone,
+    styleOptions: {
```

Comment:
These should be flat, not inside an object. Otherwise - at least the way it's currently implemented - you'd have to supply the entire object with all of the default values.

Either
* you merge the custom style options with the default style options or 
* you flatten the props

I'm in favor of flattening:
* makes the props' shape simpler
* removes the necessity to merge options
* looks cleaner when you need to change a single style only.
* Some of the properties are not "styles" in the sense of css, so the name could be confusing
* The flat approach is what the rest of the code base is using (e.g. the [`<Box />`](https://github.com/dhis2/ui/blob/master/components/box/src/box.js#L38:L50))

Reply:
I think it probably makes sense to only allow the minimal style options (`cellSize` and `width` as you mention above). Customizing the rest of the values should be handled by theme values/tokens if we want to do something like that in future, rather than overriding via props. So I vote to remove all non-needed style options.